### PR TITLE
[DOC] Change to handling of aliases

### PR DESCRIPTION
**Pull request by BurdetteLamar**
RDoc now automatically mentions aliases, so no need to mention explicitly.

### CODE REVIEW

File: doc/contributing/documentation_guide.md [hunk 1]
```diff
@@ -328,7 +327,7 @@ Return types:
 
 Aliases:
 
-- Omit aliases from the `call-seq`, but mention them near the end (see below).
+- Omit aliases from the `call-seq`.
```
Comment (jeremyevans) [inline doc/contributing/documentation_guide.md:330]: I would clarify that in rare circumstances aliases can be listed in the call-seq.  @zverok provided an example for `String#-@` and `String#dedup`.  In this case (operator and normal method are aliases), it probably makes sense to list both.  In this case, the operator and method can both be listed.

**Review by jeremyevans — CHANGES_REQUESTED** [review]
Looks good, but I think we should clarify one point.

File: doc/contributing/documentation_guide.md (hunk 1 above)
Comment (BurdetteLamar) [reply doc/contributing/documentation_guide.md:330]: I'm not sure how to characterize the exceptional cases.  Would you please add something to the doc?

**Review by jeremyevans — APPROVED** [review]
This is fine for now.  I'll update the guide to mention handling exceptional cases later.

**Pull request merged**